### PR TITLE
Refactor statuses

### PR DESCRIPTION
**Pull request by max-zilla**
- use terrautils code for generating dataset + collection mappings using exact_match flag to better support parallel instances
- add a RETRY status update for recoverable errors (e.g. Clowder down)
- check RETRY loop (formerly ERROR) with newest first

### CODE REVIEW

File: scripts/globusmonitor/globus_uploader_service.py [hunk 1]
```diff
@@ -397,22 +265,23 @@ def notifyClowderOfCompletedTask(task):
...
+        clowder_user = userMap[globUser]['clowder_user']
+        clowder_pass = userMap[globUser]['clowder_pass']
+        clowder_id = userMap[globUser]['clowder_id']
+        clowder_context = userMap[globUser]['context']
 
         sess = requests.Session()
-        sess.auth = (clowderUser, clowderPass)
+        sess.auth = (clowder_user, clowder_pass)
 
         # This will be false if any files in the task have errors; task will be revisited
```
Comment (craig-willis) [inline scripts/globusmonitor/globus_uploader_service.py]: This comment is no longer valid.

File: scripts/globusmonitor/globus_uploader_service.py [hunk 2]
```diff
@@ -67,18 +61,6 @@ def safeCopy(obj):
 
     return newObj
 
-"""If metadata keys have periods in them, Clowder will reject the metadata"""
```
Comment (craig-willis) [inline scripts/globusmonitor/globus_uploader_service.py]: Why was this removed?

File: scripts/globusmonitor/globus_uploader_service.py [hunk 3]
```diff
@@ -488,20 +383,20 @@ def notifyClowderOfCompletedTask(task):
...
-                        dsmd = sess.post(clowderHost+"/api/datasets/"+dsid+"/metadata.jsonld",
+                        dsmd = sess.post(clowder_host+"/api/datasets/"+dsid+"/metadata.jsonld",
                                          headers={'Content-Type':'application/json'},
                                          data=json.dumps(md))
 
                         if dsmd.status_code != 200:
-                            logger.error("- cannot add dataset metadata (%s: %s)" % (dsmd.status_code, dsmd.text))
-                            return False
+                            logger.error("[%s] failed to attach metadata (%s: %s)" % (ds, dsmd.status_code, dsmd.text))
+                            response = "RETRY"
```
Comment (craig-willis) [inline scripts/globusmonitor/globus_uploader_service.py]: Are you confident that all non-200 errors are retryable? Based on my experience, I'd limit to 502/504.

File: scripts/globusmonitor/globus_uploader_service.py (hunk 2 above)
Comment (max-zilla) [reply scripts/globusmonitor/globus_uploader_service.py]: our terrautils md cleaning code should perform this, it's code that should be done anywhere we read json metadata from file (e.g. cleaner extractor) and right now it's just here. hasn't been an issue since it only applies IIRC to early 2016 md files.

https://github.com/terraref/terrautils/pull/48 Here's what I'm thinking.
